File: frontend/src/app/core/dtos/booking.dto.ts
import { ResourceDto } from './resource.dto';

export interface BookingResponseDto {
  bookingId: number;
  bookingNumber: string;
  paymentId: number;
}

export interface BookingDto {
  bookingId: number;
  bookingGroupId: number;
  bookingNumber: string;
  bookingStatus: string;
  paymentId: number;
  modificationStatus: string;
  refundStatus: string;
  startedAt: Date | null;
  endedAt: Date | null;
  createdAt: Date | null;
  resources: ResourceDto[];
  totalPrice: number;
  userId: number;
}

export interface PaymentIntentDto {
  bookingId: number;
  paymentId: number;
  paymentMethodId: string;
}

File: booking-service/src/main/java/com/example/springboot/dto/BookingDto.java
package com.example.springboot.dto;

import com.example.springboot.model.BookingStatus;
import com.example.springboot.model.ModificationStatus;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.util.Date;
import java.util.List;
import java.util.UUID;

@Setter
@Getter
public class BookingDto {
    private Long bookingId;
    private UUID bookingGroupId;
    private String bookingNumber;
    private List<ResourceDto> resources;
    private Long paymentId;
    private BookingStatus bookingStatus;
    private ModificationStatus modificationStatus;
    private Date startedAt;
    private Date endedAt;
    private Date createdAt;
    private BigDecimal totalPrice;
    private Long userId;
}
